extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/WithSpanInterceptor.java

```diff
@@ -72,19 +78,39 @@ public Object span(final InvocationContext invocationContext) throws Exception {
         }
     }
 
-    private static SpanKind spanKindFromMethod(Method method) {
-        WithSpan annotation = method.getDeclaredAnnotation(WithSpan.class);
-        if (annotation == null) {
+    private static SpanKind spanKindFromMethod(Set<Annotation> annotations) {
+        SpanKind spanKind = null;
+        for (Annotation annotation : annotations) {
+            if (annotation.annotationType().getName().equals(WithSpan.class.getName())) {
+                try {
+                    spanKind = (SpanKind) annotation.annotationType().getMethod("kind").invoke(annotation);
+                } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+                break;
+            }
+        }
+        if (spanKind == null) {
             return SpanKind.INTERNAL;
         }
-        return annotation.kind();
+        return spanKind;
     }
 
     private static final class MethodRequestSpanNameExtractor implements SpanNameExtractor<MethodRequest> {
         @Override
         public String extract(final MethodRequest methodRequest) {
-            WithSpan annotation = methodRequest.getMethod().getDeclaredAnnotation(WithSpan.class);
-            String spanName = annotation.value();
+            String spanName = null;
+
+            for (Annotation annotation : methodRequest.getAnnotationBindings()) {
+                if (annotation.annotationType().getName().equals(WithSpan.class.getName())) {
+                    try {
+                        spanName = (String) annotation.annotationType().getMethod("value").invoke(annotation);
+                    } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+                        throw new RuntimeException(e);
+                    }
+                    break;
+                }
+            }
```

Comment:
Same here, using reflection here seems totally unnecessary?

Reply:
you are right